internal/controller/reinstall/reinstall_test.go

```diff
@@ -0,0 +1,680 @@
+		Expect(err).To(HaveOccurred())
+		Expect(cierrors.IsDeletionTimeoutError(err)).To(BeTrue())
+		Expect(result.Requeue).To(BeTrue())
+	})
+
+	It("should return an error if deletion fails", func() {
+		setReinstallStatusCondition(clusterInstance, *reinstallRenderedManifestsDeletedConditionStatus(metav1.ConditionFalse, v1alpha1.InProgress, "Deletion in progress"))
+
+		errorMsg := "inject error"
+		c = fakeclient.NewClientBuilder().
+			WithScheme(scheme.Scheme).
+			WithStatusSubresource(&v1alpha1.ClusterInstance{}).
+			WithObjects(clusterInstance, object).
+			WithInterceptorFuncs(interceptor.Funcs{
+				Delete: func(ctx context.Context, fclient client.WithWatch, obj client.Object, opts ...client.DeleteOption) error {
+					return errors.New(errorMsg)
+				},
+			}).
+			Build()
+
+		handler = &ReinstallHandler{
+			ConfigStore:     configStore,
+			Client:          c,
+			DeletionHandler: &deletion.DeletionHandler{Client: c, Logger: testLogger},
+			Logger:          testLogger,
+		}
+
+		result, err := handler.ensureRenderedManifestsAreDeleted(ctx, testLogger, clusterInstance)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring(errorMsg))
+		Expect(result.Requeue).To(BeTrue())
+	})
+
+	It("should requeue if deletion is still in progress", func() {
+		setReinstallStatusCondition(clusterInstance, *reinstallRenderedManifestsDeletedConditionStatus(metav1.ConditionFalse, v1alpha1.InProgress, "Deletion in progress"))
+
+		c = fakeclient.NewClientBuilder().
+			WithScheme(scheme.Scheme).
+			WithStatusSubresource(&v1alpha1.ClusterInstance{}).
+			WithObjects(clusterInstance, object).
+			Build()
+
+		handler = &ReinstallHandler{
+			ConfigStore:     configStore,
+			Client:          c,
+			DeletionHandler: &deletion.DeletionHandler{Client: c, Logger: testLogger},
+			Logger:          testLogger,
+		}
+
+		result, err := handler.ensureRenderedManifestsAreDeleted(ctx, testLogger, clusterInstance)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(result.RequeueAfter).ToNot(BeZero())
+	})
+
+	It("should complete successfully if deletion is done", func() {
+		setReinstallStatusCondition(clusterInstance, *reinstallRenderedManifestsDeletedConditionStatus(metav1.ConditionFalse, v1alpha1.InProgress, "Deletion in progress"))
+
+		c = fakeclient.NewClientBuilder().
+			WithScheme(scheme.Scheme).
+			WithStatusSubresource(&v1alpha1.ClusterInstance{}).
+			WithObjects(clusterInstance, object).
+			Build()
+
+		handler = &ReinstallHandler{
+			ConfigStore:     configStore,
+			Client:          c,
+			DeletionHandler: &deletion.DeletionHandler{Client: c, Logger: testLogger},
+			Logger:          testLogger,
+		}
+
+		// Invoke deletion first time - to delete the resource, which triggers a requeue
+		result, err := handler.ensureRenderedManifestsAreDeleted(ctx, testLogger, clusterInstance)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(result.Requeue).To(BeFalse())
```

Comment:
Your comment says this should trigger a requeue, but this says requeue should be false, and the next one says it should be true ... seems backwards?

Reply:
Fixed.